docs/csharp/tutorials/upgrade-to-nullable-references.md

```diff
@@ -0,0 +1,160 @@
+
+## Upgrade the projects to C# 8
+
+A good first step is to determine the scope of the migration task. Start by upgrading the project to C# 8.0 (or newer). Add the `LangVersion` element to both csproj files for the web project and the unit test project:
+
+```xml
+<LangVersion>8.0</LangVersion>
+```
+
+Upgrading the language version selects C# 8.0, but does not enable the nullable annotation context or the nullable warning context. Rebuild the project to ensure that it builds without warnings.
+
+A good next step is to turn on the nullable annotation context and see how many warnings are generated.  Add the following element to both csproj files in the solution, directly under the `LangVersion` element:
+
+```xml
+<NullableContextOptions>enable</NullableContextOptions>
+```
+
+Do a test build, and notice the warning list. In this small application, the compiler generates five warnings, so it's likely you'd leave the nullable annotation context enabled and start fixing warnings for the entire project.
+
+That strategy works only for the smallest projects. For any larger projects, the number of warnings generated by enabling the nullable annotation context for the entire codebase makes it harder to fix the warnings systematically. For larger enterprise projects, you'll often want to migrate one project at a time. In each project, you'll migrate one class or file at a time.
+
+## Warnings help discover original design intent
+
+There are two classes that generate multiple warnings. Start with the `NewsStoryViewModel` class. Remove the `NullableContextOptions` element from both csproj files so that you can limit the scope of warnings to the sections of code you're working with. Open the *NewsStoryViewModel.cs* file and add the following directives to enable the nullable annotation context for the `NewsStoryViewModel` and restore it following that class definition:
+
+```csharp
+#nullable enable
+public class NewsStoryViewModel
+{
+    public DateTimeOffset Published { get; set; }
+    public string Title { get; set; }
+    public string Uri { get; set; }
+}
+#nullable restore
+```
+
+These two directives help you focus your migration efforts. The nullable warnings are generated for the area of code you're actively working on. You'll leave them on until you're ready to turn on the warnings for the entire project. You should use the `restore` rather than `disable` value so that you don't accidentally disable the context later when you've turned on nullable annotations for the entire project. Once you've turned on the nullable annotation context for the entire project, you can remove all the `#nullable` pragmas from that project.
+
+The `NewsStoryViewModel` class is a data transfer object (DTO) and two of the properties are read/write strings:
+
+[!code-csharp[InitialViewModel](~/samples/csharp/tutorials/nullable-reference-migration/start/SimpleFeedReader/ViewModels/NewsStoryViewModel.cs#StarterViewModel)]
```

Comment:
This is empty:
![image](https://user-images.githubusercontent.com/12971179/53044452-e620bd00-343f-11e9-9f8a-79aa9c425c96.png)